app/pages/sections/supplier/supplier/ui.test.js

```diff
@@ -0,0 +1,125 @@
+  if (withSessionState) await setSessionState();
+};
+
+
+const getLocation = ClientFunction(() => document.location.href);
+
+fixture('Supplier page')
+  .page('http://localhost:1234/some-fake-page')
+  .afterEach(async (t) => {
+    const isDone = nock.isDone();
+    if (!isDone) {
+      nock.cleanAll();
+    }
+
+    await t.expect(isDone).ok('Not all nock interceptors were used!');
+  });
+
+test('when user is not authenticated - should navigate to the identity server login page', async (t) => {
+  await pageSetup(t);
+  nock('http://identity-server')
+    .get('/login')
+    .reply(200);
+
+  await t.navigateTo(pageUrl);
+
+  await t
+    .expect(getLocation()).eql('http://identity-server/login');
+});
+
+test('should render Supplier page', async (t) => {
+  await pageSetup(t, true, true);
+  await t.navigateTo(pageUrl);
+  const page = Selector('[data-test-id="supplier-page"]');
+
+  await t
+    .expect(page.exists).ok();
+});
+
+test('should navigate to /organisation/order-1/supplier/search/select when click on backlink', async (t) => {
+  await pageSetup(t, true, true);
+  await t.navigateTo(pageUrl);
+
+  const goBackLink = Selector('[data-test-id="go-back-link"] a');
+
+  await t
+    .expect(goBackLink.exists).ok()
+    .click(goBackLink)
+    .expect(getLocation()).eql('http://localhost:1234/order/organisation/order-1/supplier/search/select');
+});
+
+test('should render the title', async (t) => {
+  await pageSetup(t, true, true);
+  await t.navigateTo(pageUrl);
+
+  const title = Selector('h1[data-test-id="supplier-page-title"]');
+
+  await t
+    .expect(title.exists).ok()
+    .expect(await extractInnerText(title)).eql(`${content.title} order-1`);
+});
+
+test('should render the description', async (t) => {
+  await pageSetup(t, true, true);
+  await t.navigateTo(pageUrl);
+
+  const description = Selector('h2[data-test-id="supplier-page-description"]');
+
+  await t
+    .expect(description.exists).ok()
+    .expect(await extractInnerText(description)).eql(content.description);
+});
+
+test('should render the inset advice', async (t) => {
+  await pageSetup(t, true, true);
+  await t.navigateTo(pageUrl);
+
+  const insetAdvice = Selector('[data-test-id="supplier-page-insetAdvice"]');
+
+  await t
+    .expect(insetAdvice.exists).ok()
+    .expect(await extractInnerText(insetAdvice)).contains(content.insetAdvice);
+});
+
+test('should render the "Save and return" button', async (t) => {
+  await pageSetup(t, true, true);
+  await t.navigateTo(pageUrl);
+
+  const button = Selector('[data-test-id="save-button"] button');
+
+  await t
+    .expect(button.exists).ok()
+    .expect(await extractInnerText(button)).eql(content.saveButtonText);
+});
+
+
+test('should redirect to /organisation/order-1/supplier/search if no supplierSelected', async (t) => {
+  await pageSetup(t, true, false);
+  await t.navigateTo(pageUrl);
+
+  await t
+    .expect(getLocation()).eql('http://localhost:1234/order/organisation/order-1/supplier/search');
```

Comment:
should there be a check for the error?

Reply:
Not sure what you mean here?

Reply:
I actually don't understand what this test is doing

Reply:
ah ok. it should redirect to the search route if a supplierSelected from the state is not present. Doesn't throw an error

Reply:
can you just add that to the description please?